<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PluginsCenter</class>
 <widget class="QWidget" name="PluginsCenter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>758</width>
    <height>487</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="topFrame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="ClearableLineEdit" name="searchEdit">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Apply filter on plugin names</string>
        </property>
        <property name="placeholderText">
         <string>Search plugins</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="mainLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="leftLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <item>
        <widget class="QListWidget" name="pluginsSideList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>150</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <item>
          <property name="text">
           <string>All</string>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="icon">
           <iconset resource="../../../library/talipot-gui/resources/TalipotGUIResource.qrc">
            <normaloff>:/talipot/gui/icons/logo32x32.png</normaloff>:/talipot/gui/icons/logo32x32.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Algorithms</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_algorithm.png</normaloff>:/talipot/app/icons/64/plugin_algorithm.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Import/Export</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_import_export.png</normaloff>:/talipot/app/icons/64/plugin_import_export.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Glyphs</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_glyph.png</normaloff>:/talipot/app/icons/64/plugin_glyph.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Views</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_view.png</normaloff>:/talipot/app/icons/64/plugin_view.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Interactors</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_interactor.png</normaloff>:/talipot/app/icons/64/plugin_interactor.png</iconset>
          </property>
         </item>
         <item>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/64/plugin_controller.png</normaloff>:/talipot/app/icons/64/plugin_controller.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Errors</string>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/talipot/app/icons/16/error.png</normaloff>:/talipot/app/icons/16/error.png</iconset>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsUserCheckable</set>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QStackedWidget" name="pluginsContent">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="homePage">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="mainTitle">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>&lt;body&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;img src=&quot;:/talipot/gui/ui/list_bullet_arrow.png&quot; /&gt;&lt;span style=&quot;color:#626262; font-size:large;&quot;&gt;   Manage your plugins&lt;/span&gt;&lt;/p&gt;

&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:medium;&quot;&gt; &lt;/p&gt;

&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot;color:#000000;&quot;&gt;Extend Talipot functionalities and stay up-to-date by browsing and downloading the latest plugins from our repositories.&lt;/span&gt;&lt;/p&gt;
&lt;/body&gt;

&lt;/html&gt;</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="browseLayout">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QFrame" name="pluginsListTitleFrame">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="pluginsListTitle">
                <property name="text">
                 <string>&lt;b&gt;Browse categories&lt;/b&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="pluginsListFrame">
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QToolButton" name="browseAlgorithmsButton">
                <property name="toolTip">
                 <string>Algorithm plugins copmpute metrics on graph elements and modify the graph structure</string>
                </property>
                <property name="text">
                 <string>Algorithms</string>
                </property>
                <property name="icon">
                 <iconset>
                  <normaloff>:/talipot/app/icons/64/plugin_algorithm.png</normaloff>:/talipot/app/icons/64/plugin_algorithm.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>64</width>
                  <height>64</height>
                 </size>
                </property>
                <property name="toolButtonStyle">
                 <enum>Qt::ToolButtonTextUnderIcon</enum>
                </property>
                <property name="autoRaise">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QToolButton" name="browseViewsButton">
                <property name="text">
                 <string>Views</string>
                </property>
                <property name="icon">
                 <iconset>
                  <normaloff>:/talipot/app/icons/64/plugin_view.png</normaloff>:/talipot/app/icons/64/plugin_view.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>64</width>
                  <height>64</height>
                 </size>
                </property>
                <property name="toolButtonStyle">
                 <enum>Qt::ToolButtonTextUnderIcon</enum>
                </property>
                <property name="autoRaise">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="0" column="4">
               <widget class="QToolButton" name="browseInteractorsButton">
                <property name="text">
                 <string>Interactors</string>
                </property>
                <property name="icon">
                 <iconset>
                  <normaloff>:/talipot/app/icons/64/plugin_interactor.png</normaloff>:/talipot/app/icons/64/plugin_interactor.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>64</width>
                  <height>64</height>
                 </size>
                </property>
                <property name="toolButtonStyle">
                 <enum>Qt::ToolButtonTextUnderIcon</enum>
                </property>
                <property name="autoRaise">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QToolButton" name="browseGlyphsButton">
                <property name="text">
                 <string>Glyphs</string>
                </property>
                <property name="icon">
                 <iconset>
                  <normaloff>:/talipot/app/icons/64/plugin_glyph.png</normaloff>:/talipot/app/icons/64/plugin_glyph.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>64</width>
                  <height>64</height>
                 </size>
                </property>
                <property name="toolButtonStyle">
                 <enum>Qt::ToolButtonTextUnderIcon</enum>
                </property>
                <property name="autoRaise">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QToolButton" name="browseImportExportButton">
                <property name="text">
                 <string>Import/Export</string>
                </property>
                <property name="icon">
                 <iconset>
                  <normaloff>:/talipot/app/icons/64/plugin_import_export.png</normaloff>:/talipot/app/icons/64/plugin_import_export.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>64</width>
                  <height>64</height>
                 </size>
                </property>
                <property name="toolButtonStyle">
                 <enum>Qt::ToolButtonTextUnderIcon</enum>
                </property>
                <property name="autoRaise">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="pluginsListPage">
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QScrollArea" name="pluginsSearchList">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>20</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="pluginsSearchListContent">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>592</width>
              <height>436</height>
             </rect>
            </property>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="errorsPage">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>&lt;p&gt;&lt;img src=&quot;:/talipot/gui/ui/list_bullet_arrow.png&quot; /&gt;&lt;span style=&quot;color:#626262; font-size:large;&quot;&gt;   Fix plugins errors&lt;/span&gt;&lt;/p&gt;

&lt;p&gt;&lt;span style=&quot;color:#dc4d4f;&quot;&gt;Errors have been detected while loading plugins. Check the error log for details.&lt;/span&gt;&lt;/p&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QScrollArea" name="errorsLogArea">
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="errorsLogAreaContent">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>98</width>
              <height>28</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="errorsLogAreaLayout"/>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClearableLineEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">talipot/ClearableLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/TalipotApp.qrc"/>
  <include location="../../../library/talipot-gui/resources/TalipotGUIResource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>browseAlgorithmsButton</sender>
   <signal>clicked()</signal>
   <receiver>PluginsCenter</receiver>
   <slot>searchAlgorithms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseGlyphsButton</sender>
   <signal>clicked()</signal>
   <receiver>PluginsCenter</receiver>
   <slot>searchGlyphs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>413</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseImportExportButton</sender>
   <signal>clicked()</signal>
   <receiver>PluginsCenter</receiver>
   <slot>searchImportExport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseInteractorsButton</sender>
   <signal>clicked()</signal>
   <receiver>PluginsCenter</receiver>
   <slot>searchInteractors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>603</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseViewsButton</sender>
   <signal>clicked()</signal>
   <receiver>PluginsCenter</receiver>
   <slot>searchViews()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>PluginsCenter</receiver>
   <slot>setNameFilter(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pluginsSideList</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>PluginsCenter</receiver>
   <slot>sideListRowChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showErrorsPage()</slot>
  <slot>showWelcomePage()</slot>
  <slot>searchAll()</slot>
  <slot>showRepositoriesPage()</slot>
  <slot>searchImportExport()</slot>
  <slot>searchAlgorithms()</slot>
  <slot>searchGlyphs()</slot>
  <slot>searchViews()</slot>
  <slot>searchInteractors()</slot>
  <slot>setNameFilter(QString)</slot>
  <slot>sideListRowChanged(int)</slot>
  <slot>testingChecked(bool)</slot>
  <slot>stableChecked(bool)</slot>
  <slot>repoAdded()</slot>
  <slot>repoRemoved()</slot>
 </slots>
</ui>
